### PR TITLE
fix(calendarEvent): #MC-297 disable save button when too old start date and too far end date

### PR DESCRIPTION
**Pull request by Gregprod33**
## Describe your changes
To prevent user from saving too long events and slow down the application, we now display a message and disable save button if the start date of event is too old (before january 1, 2000) or end date is more than in 80 years.
## Checklist tests
Create a new event and choose a date before 01 01 2000 or a end date after 2103
## Issue ticket number and link
[MC-297](https://jira.support-ent.fr/browse/MC-297)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/controllers/controller.ts [hunk 1]
```diff
@@ -85,6 +85,8 @@ export const calendarController = ng.controller('CalendarController',
             $scope.ENABLE_RBS = ENABLE_RBS;
             $scope.rbsEmitter = new RbsEmitter($scope, !!$scope.ENABLE_RBS);
             $scope.ENABLE_ZIMBRA = ENABLE_ZIMBRA;
+            $scope.minDate = moment('2000-01-01');
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:88]: Mets '2000-01-01' et 80 dans le fichier de const TIME.ts

File: src/main/resources/public/ts/controllers/controller.ts [hunk 2]
```diff
@@ -1662,7 +1665,7 @@ export const calendarController = ng.controller('CalendarController',
                 $scope.eventForm = angular.element(document.getElementById("event-form")).scope();
                 /** Ensures that the fields of the form are correctly filled*/
                 let areFieldsInCommonValid: boolean = ($scope.rbsEmitter.checkBookingValidAndSendInfoToSniplet() && !$scope.eventForm.editEvent.$invalid && $scope.isCalendarSelectedInEvent()
-                    && $scope.isTimeValid() && $scope.isDateValid() &&  $scope.areRecurrenceAndEventLengthsCompatible());
+                    && $scope.isTimeValid() && $scope.isDateValid() &&  $scope.areRecurrenceAndEventLengthsCompatible() && !$scope.isStartDateToOld() && !$scope.isEndDateToFar());
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1668]: il se passe quoi si on fait une requete postman avec un évent trop long ? --> vérif back en + peut-être ?

**Review by alicedraillard — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/controllers/controller.ts [hunk 3]
```diff
@@ -1699,6 +1702,16 @@ export const calendarController = ng.controller('CalendarController',
                 && moment($scope.calendarEvent.startMoment).isValid() && moment($scope.calendarEvent.endMoment).isValid()
                 && moment($scope.calendarEvent.startMoment).isSameOrBefore(moment($scope.calendarEvent.endMoment), 'day'));
 
+            /**
+             * Returns true if startMoment is older than january 1/2000
+             */
+            $scope.isStartDateToOld = (): boolean => ($scope.calendarEvent.startMoment < $scope.minDate);
```
Comment (qperie) [inline src/main/resources/public/ts/controllers/controller.ts:1708]: Juste une histoire d'anglais mais c'est "isStartDateTooOld" avec 2 "oo" (3 avec celui de "Old")

File: src/main/resources/public/ts/controllers/controller.ts [hunk 4]
```diff
@@ -1699,6 +1702,16 @@ export const calendarController = ng.controller('CalendarController',
...
+
+            /**
+             * Returns true if endMoment is after 80 years further
+             */
+            $scope.isEndDateToFar = (): boolean => ($scope.calendarEvent.endMoment > $scope.maxDate);
```
Comment (qperie) [inline src/main/resources/public/ts/controllers/controller.ts:1713]: Même remarque qu'au-dessus

File: src/main/resources/public/ts/controllers/controller.ts (hunk 2 above)
Comment (qperie) [reply src/main/resources/public/ts/controllers/controller.ts:1668]: Très bonne remarque

**Review by qperie — APPROVED** [review]

File: src/main/resources/public/ts/controllers/controller.ts [hunk 5]
```diff
@@ -1699,6 +1702,30 @@ export const calendarController = ng.controller('CalendarController',
...
+             * Returns true if endMoment is after 80 years further
+             */
+            $scope.isEndDateTooFar = (): boolean => ($scope.calendarEvent.endMoment > $scope.maxDate);
+
+            /**
+             * Check the end date of recurrence
+             */
+            $scope.isValidRecurrentEndDate = (): boolean => {
+                const { end_type, end_on } = $scope.calendarEvent.recurrence;
+                if (end_type === 'on' && end_on) {
```
Comment (FlorentMr) [inline src/main/resources/public/ts/controllers/controller.ts:1720]: Vérification utile? (pas sûr de comprendre de pourquoi recurrence peut changer de type, j'ai pas le contexte)

**Pull request merged**